src/fusor/translator.py

```diff
@@ -0,0 +1,580 @@
+    to FUSOR AssayedFusion objects
+    """
+
+    def __init__(self, fusor: FUSOR) -> None:
+        """Initialize Translator class
+
+        :param fusor: A FUSOR instance
+        """
+        self.fusor = fusor
+
+    def _format_fusion(
+        self,
+        gene_5prime: GeneElement,
+        gene_3prime: GeneElement,
+        tr_5prime: TranscriptSegmentElement | None = None,
+        tr_3prime: TranscriptSegmentElement | None = None,
+        ce: CausativeEvent | None = None,
+        rf: bool | None = None,
+        assay: Assay | None = None,
+    ) -> AssayedFusion:
+        """Format classes to create AssayedFusion objects
+
+        :param gene_5prime: 5'prime GeneElement
+        :param gene_3prime: 3'prime GeneElement
+        :param Optional[tr_5prime]: 5'prime TranscriptSegmentElement
+        :param Optional[tr_3prime]: 3'prime TranscriptSegmentElement
+        :param Optional[ce]: CausativeEvent
+        :param Optional[rf]: A boolean indicating if the reading frame is preserved
+        :param Optional[assay]: Assay
+        :return AssayedFusion object
+        """
+        params = {}
+        if not tr_5prime[0] and not tr_3prime[0]:
+            params["structure"] = [gene_5prime, gene_3prime]
+        elif tr_5prime[0] and not tr_3prime[0]:
+            params["structure"] = [tr_5prime[0], gene_3prime]
+        elif not tr_5prime[0] and tr_3prime[0]:
+            params["structure"] = [gene_5prime, tr_3prime[0]]
+        else:
+            params["structure"] = [tr_5prime[0], tr_3prime[0]]
+
+        if ce:
+            params["causativeEvent"] = ce
+        if rf:
+            params["readingFramePreserved"] = rf
+        if assay:
+            params["assay"] = assay
+        return AssayedFusion(**params), None
+
+    def _get_causative_event(
+        self, chrom1: str, chrom2: str, descr: str | None = None
+    ) -> CausativeEvent | None:
+        """Infer Causative Event. Currently restricted to rearrangements
+
+        :param chrom1: The chromosome for the 5' partner
+        :param chrom2: The chromosome for the 3' partner
+        :param descr: An annotation describing the fusion event. This input is supplied to the eventDescription CausativeEvent attribute.
+        :return: A CausativeEvent object if construction is successful
+        """
+        if descr and "rearrangement" in descr:
+            return CausativeEvent(
+                eventType=EventType("rearrangement"), eventDescription=descr
+            )
+        if chrom1 != chrom2:
+            return CausativeEvent(eventType=EventType("rearrangement"))
+        return None
+
+    def _get_gene_element_unnormalized(self, symbol: str) -> GeneElement:
+        """Return GeneElement when gene symbol cannot be normalized
+
+        :param symbol: A gene symbol for a fusion partner
+        :return: A GeneElement object
+        """
+        return GeneElement(
+            type="GeneElement",
```

Comment:
I wouldn't worry about providing the `type` param. In VRS and VRS-y classes, the `type` is almost always both provided by default and required to match that default, so there's no need to pass it explicitly like this